### ISSUE TITLE
Opencl/Cuda support

### ISSUE THREAD

**Issue by Titaniumtown** [rust-ml/linfa#92]
I've been recently getting into rust and I do a lot with machine learning (in python with tensorflow), so I was wondering if there was any opencl or cuda support by any chance? I was looking in the readme and it didn't seem to mention anything, only blas implementations in intel mkl.

**Comment by bytesnake** [comment]
Probably not, at least not anytime soon .. this is at least blocked on https://github.com/rust-ndarray/ndarray/issues/840 and then we have to re-implement many of the LAPACK routines in GPU accelerated versions. This is just not feasible with the current size/interest in the ecosystem. We are similar to `scikit-learn` in scope and application.

If you want a neural network+AD library, à la tensorflow, then take a look at [tch-rs](https://github.com/LaurentMazare/tch-rs) which interfaces with Torch.

**Comment by bytesnake** [comment]
I will close this because it is really out of scope

**Comment by bytesnake** [comment]
but we should add a section to the README explaining the situation! Will do this later this day

**Comment by spihere** [comment]
We now have a new repo that implements CUDA in rust [here](https://github.com/Rust-GPU/Rust-CUDA). I can help implementing this if issue is reopened.

**Comment by YuhanLiin** [comment]
The Rust-CUDA repo is still in the experimental stage and the `ndarray` GPU integration ticket is still open, so this is still a no-go. We should really add this explanation to the README though.
